### DISCUSSION TITLE
Standlone output + Dynamic Routes results in 404 of /index

### DISCUSSION THREAD

**Discussion by wangpin34** [vercel/next.js#37478] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Home China
    Binaries:
      Node: 16.15.0
      npm: 8.5.5
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.1.7-canary.29
      react: 18.1.0
      react-dom: 18.1.0

### What browser are you using? (if relevant)

Chrome 102.0.5005.63

### Describe the Bug

I use Dynamic Routes to generate pages for the docs. In the dev mode it works fine and the `index.md` will be generated as page at path `/index`. The `start` command works as well. 

The problem is when I build with `outputStandalone` enabled, the `/index` shows `404` when I started the service with `node .next/standalone/server.js`. 

here is my `next.config.js`.
```
/** @type {import('next').NextConfig} */
const nextConfig = {
  experimental: {
    outputStandalone: true,
  },
  reactStrictMode: true,
}

module.exports = nextConfig
```

The minimal reproducible example was uploaded here: https://github.com/wangpin34/next-standalone

### Expected Behavior

The `/index` page is loaded successfully. 

### To Reproduce

Download the minimal reproducible example from https://github.com/wangpin34/next-standalone.  

Install and build
```
npm install
npm run build
```

Then start the server
```
node .next/standalone/server.js
```

Finally open http://localhost:3000/index on your browser.

**Comment by balazsorban44** [comment, marked as answer]
The issue is that `/` is not caught by the `[...doc]` route.

You will need to use the optional catch-all route form `[[...doc]]` if you want to handle `/` the same way. (remember to set the `params.doc` value to `["index"]` by default in your `getStaticProps`).

See the docs: https://nextjs.org/docs/routing/dynamic-routes#optional-catch-all-routes


As a suggestion, you might want to check out our official Markdown guide to simplify your code. I.e. You can define markdown files directly under `pages` and Next.js will load them as regular pages. Check out: https://nextjs.org/docs/advanced-features/using-mdx

**Reply by wangpin34** [reply]
@balazsorban44  Thanks for your reply.  The `optional-catch-all-routes` does resolve the issue. 

However I am still confused about why the `/index` page was not found.  You mentioned 

> The issue is that / is not caught by the [...doc] route.

I was not going to view the `/` but the `/index`, in this case the `/index` page should be available just as the dev or start mode(next, next start). However unfortunatly it didn't work using the standalone output.  May be I miss someting important regarding to rules of `index` route. It would be greate if you point it out for me.

Hope you have a good day.

**Reply by balazsorban44** [reply]
Sorry, at the time I thought you meant `/` when you wrote `/index`.

I'm just seeing this again, there is a similar open issue about this at: https://github.com/vercel/next.js/issues/38742

`output: "standalone"` (now out of experimental) changes the `/index` behavior slightly. You can follow the above issue, we will try to clear this up there, and release a fix if it's confirmed to be a bug and hopefully, that will be the solution for you too. :pray:
